src/Symfony/Component/Security/Http/AccessToken/Oidc/OidcTokenHandler.php

```diff
@@ -55,52 +51,49 @@ public function getUserBadgeFrom(string $accessToken): UserBadge
             throw new \LogicException('You cannot use the "oidc" token handler since "web-token/jwt-signature" and "web-token/jwt-checker" are not installed. Try running "composer require web-token/jwt-signature web-token/jwt-checker".');
         }
 
+        $jwsVerifier = new JWSVerifier(new AlgorithmManager([$this->signatureAlgorithm]));
+        $serializerManager = new JWSSerializerManager([new CompactSerializer()]);
+
         try {
-            // Decode the token
-            $jwsVerifier = new JWSVerifier(new AlgorithmManager([$this->signatureAlgorithm]));
-            $serializerManager = new JWSSerializerManager([new CompactSerializer()]);
             $jws = $serializerManager->unserialize($accessToken);
-            $claims = json_decode($jws->getPayload(), true);
+            $claims = json_decode($jws->getPayload(), true, 512, JSON_THROW_ON_ERROR);
+        } catch (\InvalidArgumentException|\JsonException $e) {
+            throw new BadCredentialsException('Unable to parse the token.', 0, $e);
+        }
 
-            // Verify the signature
-            if (!$jwsVerifier->verifyWithKey($jws, $this->jwk, 0)) {
-                throw new InvalidSignatureException();
```

Comment:
throwing specific exceptions to catch them immediately is pointless. If we want to provide extension points to allow e.g. fixing the token on the fly, we can do it by dispatching specific events.